builder/azure/arm/builder.go

```diff
@@ -371,24 +387,32 @@ func (b *Builder) getServicePrincipalTokens(say func(string)) (*adal.ServicePrin
 	var err error
 
 	if b.config.useDeviceLogin {
-		servicePrincipalToken, err = packerAzureCommon.Authenticate(*b.config.cloudEnvironment, b.config.TenantID, say)
+		servicePrincipalToken, err = packerAzureCommon.Authenticate(*b.config.cloudEnvironment, b.config.TenantID, say, b.config.cloudEnvironment.ServiceManagementEndpoint)
```

Comment:
👍

Reply:
Why do you have pass this into the method twice?  Why don't you just called b.config.cloudEnvironment.ServiceManagementEndpoint when you're in the method?

Reply:
Because we need two separate tokens for service manager as well keyvault